src/Common/src/System/Data/Locale/Locale.cs

```diff
@@ -1,58 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System.Collections.Concurrent;
-using System.Data.Common;
-using System.Globalization;
-using System.Text;
-
-namespace System.Data
-{
-    internal static class Locale
-    {
-        // Maps between LCIDs and LocaleName+AnsiCodePages+Encodings
-        private static readonly ConcurrentDictionary<int, Tuple<string, int, Encoding>> s_cachedEncodings = new ConcurrentDictionary<int, Tuple<string, int, Encoding>>();
-
-        static Locale()
-        {
-            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);            
```

Comment:
This warrants more discussion. @tarekgh also had some concerns about this. Can you open an issue for this?